daml-lf/language/src/main/scala/com/digitalasset/daml/lf/language/Ast.scala

```diff
@@ -686,9 +681,9 @@ object Ast {
       observers: E, // Observers of the contract.
       key: Option[GenTemplateKey[E]],
       implements: Map[TypeConName, GenTemplateImplements[E]],
-  ) extends NoCopy
```

Comment:
There is no special invariant for this case class, so no reason to have a private constructor and prevent copy method.